python/pyarrow/tensor.pxi

```diff
@@ -339,6 +350,15 @@ shape: {0.shape}""".format(self)
     def non_zero_length(self):
         return self.stp.non_zero_length()
 
+    @property
+    def has_canonical_format(self):
+        cdef:
+            _CSparseCOOIndexPtr csi
+
+        csi = dynamic_cast[_CSparseCOOIndexPtr](self.stp.sparse_index().get())
```

Comment:
Why the `dynamic_cast`? Since this is a sparse COO tensor, the cast should always succeed. Just use a normal Cython cast.

Reply:
The reason is that I couldn't find normal Cython's cast when I wrote this code.  Thank you for telling me.